Python/compile.c

```diff
@@ -9240,40 +9243,40 @@ basicblock_has_lineno(const basicblock *bb) {
     return false;
 }
 
-/* If this block ends with an unconditional jump to an exit block,
- * then remove the jump and extend this block with the target.
+/* If this block ends with an unconditional jump to a small exit block that does
+ * not have linenos, then remove the jump and extend this block with the target.
```

Comment:
OOI, has this always only copied blocks without line numbers, or is that a recent change? 
The name is  bit misleading, and there is no reason why blocks with line numbers cannot be copied.

Reply:
It's a recent change: bde06e1b8381f140b296a397ddd1deb1c784ff8e

Reply:
But I see that your subsequent changes make it unnecessary, so I think it can be reverted.